src/sentry/api/event_search.py

```diff
@@ -0,0 +1,167 @@
+from __future__ import absolute_import
+
+from collections import namedtuple
+from parsimonious.grammar import Grammar, NodeVisitor
+
+from sentry.search.utils import parse_datetime_string, InvalidQuery
+
+event_search_grammar = Grammar(r"""
+# raw_search must come at the end, otherwise other
+# search_terms will be treated as a raw query
+search          = search_term* raw_search?
+search_term     = space? (time_filter / basic_filter) space?
+raw_search      = ~r".+$"
+
+# standard key:val filter
+basic_filter    = search_key sep search_value
+# filter specifically for the timestamp
+time_filter     = "timestamp" operator date_formats
+
+search_key      = ~r"[a-z]*\.?[a-z]*"
+search_value    = ~r"\S*"
```

Comment:
Do we need/want to support values with spaces? Environment names and release names can contain whitespace in them.

Reply:
yeah i need to figure out how to do that exactly/what the current  behavior is. i think release might need to be a special case like timestamp since we will likely want to support prefixes like `2.1.*`.

i'll file a ticket to come back to this so i don't forget about it

Reply:
We'll definitely just need to support quoting. So quoted vs unquoted values. unquoted values have limited characters, vs quoted values support escaping, etc.

Reply:
Also with regard to releases specifically and that globbing, we support this currently through https://docs.python.org/2/library/fnmatch.html. Obviously this doesn't work in ClickHouse, but if we take the version string, it can be translated into a regular expression then probably shove that into ClickHouse. (https://docs.python.org/2/library/fnmatch.html#fnmatch.translate)